.github/workflows/add-label-when-promoted.yaml

```diff
@@ -0,0 +1,36 @@
+name: Check if commits are promoted
+
+on: [push]
+
+jobs:
+  check-commit:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    if: github.ref == 'refs/heads/master'
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Fetch all history for all tags and branches
+
+      - name: Label PRs
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # Get the list of the latest commits
+          commits=$(git log --format=%B -n 100 HEAD~100..HEAD)
```

Comment:
How about reading all commits from previous next head?

Reply:
I will try to modify it

Reply:
Changed. GitHub events have the last sha and current,so we will use those values to compare